File: app/src/main/java/farasense/mobile/service/download/DownloadFaraSenseSensorDailyService.java
package farasense.mobile.service.download;

import java.util.Date;
import java.util.List;

import farasense.mobile.api.RestClient;
import farasense.mobile.api.base.ErrorListener;
import farasense.mobile.api.base.RestError;
import farasense.mobile.api.base.SuccessListener;
import farasense.mobile.model.DAO.FaraSenseSensorDailyDAO;
import farasense.mobile.model.realm.FaraSenseSensorDaily;
import farasense.mobile.service.base.BaseService;
import farasense.mobile.service.listener.OnDownloadContentListener;

public class DownloadFaraSenseSensorDailyService extends BaseService {

    public static void download(OnDownloadContentListener onDownloadContentListener, Date startDate, Date finalDate) {

        new Thread(() -> RestClient.getInstance().getFaraSenseSensorDaily(new SuccessListener<List<FaraSenseSensorDaily>>() {
            @Override
            public void onSuccess(List<FaraSenseSensorDaily> response) {
                onDownloadContentListener.onSucess();
                FaraSenseSensorDailyDAO.saveFromServer(response);
            }
        }, new ErrorListener() {
            @Override
            public void onError(RestError restError) {

            }
        }, startDate, finalDate)).start();
    }
}

File: app/src/main/java/farasense/mobile/service/download/DownloadFaraSenseSensorDailyService.kt
package farasense.mobile.service.download

import android.annotation.SuppressLint
import android.content.Context
import java.util.Date
import farasense.mobile.api.RestClient
import farasense.mobile.api.base.ErrorListener
import farasense.mobile.api.base.RestError
import farasense.mobile.api.base.SuccessListener
import farasense.mobile.model.DAO.FaraSenseSensorDailyDAO
import farasense.mobile.model.realm.FaraSenseSensorDaily
import farasense.mobile.service.base.BaseService
import farasense.mobile.service.listener.OnDownloadContentListener

@SuppressLint("Registered")
class DownloadFaraSenseSensorDailyService : BaseService() {

    companion object {

        fun download(context: Context, onDownloadContentListener: OnDownloadContentListener, startDate: Date, finalDate: Date) {
            Thread {
                RestClient.getFaraSenseSensorDaily(context, object : SuccessListener<List<FaraSenseSensorDaily>>() {
                    override fun onSuccess(response: List<FaraSenseSensorDaily>) {
                        onDownloadContentListener.onSucess()
                        FaraSenseSensorDailyDAO().saveFromServer(response)
                    }
                }, object : ErrorListener() {
                    override fun onError(restError: RestError) {

                    }
                }, startDate, finalDate)
            }.start()
        }
    }
}
